### DISCUSSION TITLE
How to enable bluetooth with BlueR

### DISCUSSION THREAD

**Discussion by Jan-Schweizer** [bluez/bluer#141] [Q&A] [answered]
Hey,
I am building (for educational purposes only) a [KMP](https://kotlinlang.org/docs/multiplatform.html) app which targets Android and desktop (via JVM). The idea is to send and receive files via Bluetooth.
I'm using Rust JNI and BlueR for the desktop Bluetooth handling.

Is it possible to enable Bluetooth via BlueR?
Currently, I'm listening to `SessionEvent`s to inform the app whenever an adapter is added or removed. This works great when I enable/disable Bluetooth via the system. I'd also like to be able to request to enable Bluetooth from within the app.
I have seen that BlueR provides `Adapter::set_powered`. This however is not usable, since I have have no access to an adapter when Bluetooth is disabled.

I'm not sure if this is a configuration issue of my Bluetooth, or a feature request for the crate.
Thanks and kind regards

**Comment by surban** [comment, marked as answer]
bluetoothd only supports setting the power state of the adapter using `Adaper::set_powered`.

You need to find out how your system disables Bluetooth. The standard way is the [rfkill framework](https://www.kernel.org/doc/Documentation/rfkill.txt). You would need to plug into that to re-enable Bluetooth.

**Reply by Jan-Schweizer** [reply]
Thanks for the quick reply. I'll have a look into this
